### PR TITLE
Moved community data to separate file

### PR DESCRIPTION
**Pull request by Calinator444**
I moved the data for all of the local communities to a separate file and used javascript to generate a "card" for each community. This should make changes simpler because the card template only needs to be updated once.

### CODE REVIEW

**Comment by colecornford** [comment]
@clbaita pls

File: src/js/communities/communities.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      website ? website : meetUp} >${name}</a>
+  </h4>
+  <dl>
+    <dt>Links:</dt>
+    <dd>
+      <ul>
+         
+        ${linkedIn ? `<li><a href=${linkedIn} rel="external">LinkedIn</a></li>` : ""}
+        ${facebook ? `<li><a href=${facebook} rel="external">Facebook</a></li>` : ""}
+        ${meetUp ? `<li><a href=${meetUp}} rel="external">MeetUp</a></li>` : ""}
```
Comment (clbaita) [inline src/js/communities/communities.js:20]: There is an extra } here after meetup, same with twitter.

File: index.html [hunk 2]
```diff
@@ -163,152 +163,8 @@ <h1>Community</h1>
     <p>There is a bunch of niche communities in the digital, design, coding spaces. Here is our list:</p>
```
Comment (clbaita) [inline index.html:164]: I know this is super minor but could we please remove the empty lines here for consistency.

File: src/js/communities/communities.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import {Communities} from "./communities-data.js";
+const renderArea = document.getElementById("groups");
+Communities.map((val)=>{
+  const {name, location, period, links, demographic, format} = val;
+  const {twitter, facebook, website, meetUp, linkedIn} = links;
+  const {interval, day, weekDay} = period;
+
+  const newHTML = `<h4>
+    <a href=${
+      //uses the group's Meetup link if they don't have a website
```
Comment (clbaita) [inline src/js/communities/communities.js:10]: Again SUPER minor but could we polish this up with a space and capital letter. eg.
// Uses the group's Meetup link if they don't have a website

**Review by clbaita — CHANGES_REQUESTED** [review]
Hey @Calinator444 these are nice changes and thanks so much for your effort but we may need to change up your approach a little.

Rather than writing direct html and using .innerHtml lets use jsx (react) to help make sure when we print stuff to the page it is safe. Currently if any data in that file is html it will render as such creating an [XXS](https://portswigger.net/web-security/cross-site-scripting) vulnerability. For example if you replace any text in that file with something like `</img>` it will execute.

If you create it using react then the data will be output encoded and any tags like that will be printed as text not html.

This isn't a big issue because we control all the data in that file and it is hard coded but ideally we follow best practice, no reason not to be safe. It's also good to know for the future.

Take a look at the way it's done in Meetup.js in the `render` and `renderList` functions. You want to call preacts render function and pass it the document.getElement and the *jsx* component.

If you need any help feel free to reach out to me on slack and I would be happy to talk you through it.

Thanks again for taking the time to contribute.

File: src/js/communities/communities.js (hunk 3 above)
Comment (Calinator444) [reply src/js/communities/communities.js:10]: Thank you for your feedback. I'll be sure to revise my changes accordingly.

**Pull request closed without merging**